app/src/main/java/io/rapidpro/androidchannel/IncomingSMSObserver.java

```diff
@@ -61,8 +64,18 @@ public void onChange(boolean selfChange) {
 
         Uri inboxUri = Uri.parse("content://sms/inbox");
 
+        String ignoredMessage = prefs.getString(SettingsActivity.SMS_IGNORED_MESSAGE, "");
+        String selection = null;
+        String [] selectionArgs = null;
+
+        if (!TextUtils.isEmpty(ignoredMessage)) {
+            selection = "body NOT LIKE ?";
+            selectionArgs = new String[]{"%" + ignoredMessage + "%"};
```

Comment:
I would like to avoid useless iterations, but I agree that it could be done in a better way.